src/lib/core/tests/TestTLV.cpp

```diff
@@ -203,92 +200,98 @@ void ForEachElement(nlTestSuite * inSuite, TLVReader & reader, void * context,
 
 struct TestTLVContext
 {
-    nlTestSuite * mSuite   = nullptr;
     int mEvictionCount     = 0;
     uint32_t mEvictedBytes = 0;
+};
 
-    TestTLVContext(nlTestSuite * suite) : mSuite(suite) {}
+class TestTLV : public ::testing::Test
+{
+public:
+    static TestTLVContext ctx;
+    static void SetUpTestSuite() { VerifyOrDie(chip::Platform::MemoryInit() == CHIP_NO_ERROR); }
+    static void TearDownTestSuite() { chip::Platform::MemoryShutdown(); }
 };
 
-void TestNull(nlTestSuite * inSuite, TLVReader & reader, Tag tag)
+TestTLVContext TestTLV::ctx;
+
+void TestNull(TLVReader & reader, Tag tag)
 {
-    NL_TEST_ASSERT(inSuite, reader.GetType() == kTLVType_Null);
-    NL_TEST_ASSERT(inSuite, reader.GetTag() == tag);
-    NL_TEST_ASSERT(inSuite, reader.GetLength() == 0);
+    EXPECT_EQ(reader.GetType(), kTLVType_Null);
+    EXPECT_EQ(reader.GetTag(), tag);
+    EXPECT_EQ(reader.GetLength(), 0u);
 }
 
-void TestString(nlTestSuite * inSuite, TLVReader & reader, Tag tag, const char * expectedVal)
+void TestString(TLVReader & reader, Tag tag, const char * expectedVal)
 {
-    NL_TEST_ASSERT(inSuite, reader.GetType() == kTLVType_UTF8String);
-    NL_TEST_ASSERT(inSuite, reader.GetTag() == tag);
+    EXPECT_EQ(reader.GetType(), kTLVType_UTF8String);
+    EXPECT_EQ(reader.GetTag(), tag);
 
     size_t expectedLen = strlen(expectedVal);
-    NL_TEST_ASSERT(inSuite, reader.GetLength() == expectedLen);
+    EXPECT_EQ(reader.GetLength(), expectedLen);
 
     chip::Platform::ScopedMemoryBuffer<char> valBuffer;
     char * val = static_cast<char *>(valBuffer.Alloc(expectedLen + 1).Get());
 
     CHIP_ERROR err = reader.GetString(val, static_cast<uint32_t>(expectedLen) + 1);
-    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    EXPECT_EQ(err, CHIP_NO_ERROR);
 
-    NL_TEST_ASSERT(inSuite, memcmp(val, expectedVal, expectedLen + 1) == 0);
+    EXPECT_EQ(memcmp(val, expectedVal, expectedLen + 1), 0);
 }
 
-void TestDupString(nlTestSuite * inSuite, TLVReader & reader, Tag tag, const char * expectedVal)
+void TestDupString(TLVReader & reader, Tag tag, const char * expectedVal)
 {
-    NL_TEST_ASSERT(inSuite, reader.GetType() == kTLVType_UTF8String);
-    NL_TEST_ASSERT(inSuite, reader.GetTag() == tag);
+    EXPECT_EQ(reader.GetType(), kTLVType_UTF8String);
+    EXPECT_EQ(reader.GetTag(), tag);
 
     size_t expectedLen = strlen(expectedVal);
-    NL_TEST_ASSERT(inSuite, reader.GetLength() == expectedLen);
+    EXPECT_EQ(reader.GetLength(), expectedLen);
 
     char * val     = nullptr;
     CHIP_ERROR err = reader.DupString(val);
-    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
-    NL_TEST_ASSERT(inSuite, val != nullptr);
+    EXPECT_EQ(err, CHIP_NO_ERROR);
+    ASSERT_NE(val, nullptr);
     if (val != nullptr)
```

Comment:
@`if` not required here since there is an assert beforehand. There are other similar patters in the test code, so you can remove `if`s.